### ISSUE TITLE
Memory coruption when running lsd_slam_core

### ISSUE THREAD

**Issue by boonflies** [Hypha-ROS/hypharos_ardrone_navigation#1]
I receive the following error, when i run,
$ rosrun lsd_slam_core live_slam image:=/ardrone/front/image_rect camera_info:=/ardrone/front/camera_info

_Received ROS Camera Calibration: fx: 463.275726, fy: 535.977356, cx: 328.456696, cy: 134.693726 @ 640x360
RECEIVED ROS camera calibration!
image dimensions must be multiples of 16! Please crop your images / video accordingly.
Started mapping thread!
Started  constraint search thread!
Started optimization thread 
Doing Random initialization!
started image display thread!
Done Random initialization!
*** Error in `/home/dyana/ros/catkin_ws/devel/lib/lsd_slam_core/live_slam': malloc(): memory corruption: 0x0000000000cb04a0 ***
Aborted (core dumped)_

Then I tried to use a calibration file, instead of camera_info,
rosrun lsd_slam_core live_slam image:=/ardrone/front/image_rect _calib:=/home/dyana/.ros/camera_info/ardrone1_default.cfg
with calibration file as follows
0.672049 0.899033 0.502065 0.513876 0.271972
640 352
none
640 352

then i received a different error,
_UndistorterPTAM: input image size differs from expected input size! Not undistorting_

How to solve this memory corruption error? and undistortingPTAM error?

**Comment by kungfrank** [comment]
Hello @boonflies,
Memory corruption error in lsd_slam is cause by input camera resolution isn't multiple-of-16. That why we modified ardrone output image resolution to 640x352(see video.h in ardrone_autonomy, you should replace same file in your ardrone_autonomy with it ). But at the same time don't forget to modify parameter of resolution in camera calibration file too(something like ardrone_front.yaml which is ardrone's default camera calibration file).

**Comment by jim1993** [comment]
Hi @boonflies 
It is great to know you have solved the issue!
About the resolution, well, based on my experience, there is no performance difference.
The only thing you need to take care is to do the camera calibration after modifying the image size!

**Comment by boonflies** [comment]
I did not properly calibrate the camera, but I just got some calibration file from net. I am using this calibration file,
0.50355332 0.894045767 0.494061013 0.509863 -0.466013 0.176685 0.032364 -0.010129 0.000000
640 360
crop
400 288

I have not visualized the point cloud in RViz yet. I can only see it in lsd_slam's point cloud viewer.

Is it right, that we get point cloud when we press 'l' followed by 'p' on the lsd_slam's pointcloud viewer and a tmp.ply file is generated. How is the point cloud topic published and visualized in RViz?
